### PR TITLE
Setup contentgrid-spring-swagger

### PR DESCRIPTION
**Pull request by tgeens**
New small contentgrid-spring subproject with 2 goals:
* facilitate updating `swagger-ui` dependency with renovate
* get rid of the following dependencies in the ContentGrid apps `build.gradle`:
  - `org.webjars:swagger-ui`
  - `org.webjars:webjars-locator-core`

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -25,4 +25,5 @@ include 'contentgrid-spring-boot-starter-annotations'
 include 'contentgrid-spring-boot-platform'
 include 'contentgrid-spring-data-rest'
 include 'contentgrid-spring-integration-events'
-include 'contentgrid-spring-querydsl'
\ No newline at end of file
+include 'contentgrid-spring-querydsl'
+include 'contentgrid-spring-swagger'
```
Comment (vierbergenlars) [inline settings.gradle:29]:
```suggestion
include 'contentgrid-spring-swagger-ui'
```

This only supports swagger UI, not the open api spec (a long time ago called swagger spec) itself.

File: contentgrid-spring-swagger/src/main/resources/swagger-initializer.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+window.onload = function() {
+    //<editor-fold desc="Changeable Configuration Block">
+
+    // the following lines will be replaced by docker/configurator, when it runs in a docker-container
```
Comment (vierbergenlars) [inline contentgrid-spring-swagger/src/main/resources/swagger-initializer.js:4]: This is not exactly true; that's a comment coming from where the file was copied from, bus is not accurate in this project

File: contentgrid-spring-swagger/src/main/java/com/contentgrid/spring/swaggerui/SwaggerUIInitializerController.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.contentgrid.spring.swaggerui;
```
Comment (vierbergenlars) [inline contentgrid-spring-swagger/src/main/java/com/contentgrid/spring/swaggerui/SwaggerUIInitializerController.java:1]: Isn't the resource file served automatically when it's put on the classpath in the correct location?

Comment (tgeens) [reply contentgrid-spring-swagger/src/main/java/com/contentgrid/spring/swaggerui/SwaggerUIInitializerController.java:1]: It is, but in this case it is problematic:
- either you put it in a versioned folder, like /webjars/swagger-ui/4.11.1/ and you have problems when the dependency gets updated
- if you put it one level higher, the webjar version detection breaks

Comment (vierbergenlars) [reply contentgrid-spring-swagger/src/main/java/com/contentgrid/spring/swaggerui/SwaggerUIInitializerController.java:1]: That is something useful to put in a comment on the class then?

File: contentgrid-spring-swagger/src/main/resources/swagger-initializer.js (hunk 2 above)
Comment (tgeens) [reply contentgrid-spring-swagger/src/main/resources/swagger-initializer.js]: I copied it varbatim from the current project. I'll clean it up.

File: contentgrid-spring-swagger/src/main/java/com/contentgrid/spring/swaggerui/SwaggerUIInitializerController.java (hunk 3 above)
Comment (tgeens) [reply contentgrid-spring-swagger/src/main/java/com/contentgrid/spring/swaggerui/SwaggerUIInitializerController.java]: See updated javadoc

**Review by vierbergenlars — APPROVED** [review]

**Pull request merged**